### ISSUE TITLE
Debugging the sortino optimizer, sortino rate not always calculated.

### ISSUE THREAD

**Issue by robinvanleeuwen** [notadamking/RLTrader#59]
I am trying to get a grasp on how the optimizer works and which values are given/calculated etc. 
I Added some debugging, and see that the sortino rate is not always calculated. 
This happens at least when the first couple of steps are non-trades. See log below.

I also have a question about the rate:
the code specifies that  only finite rates are measured. That is NaN and "infinite".
NaN i assume the rate cannot be calculated. But infinite is also set to zero. Is this correct?
What does an calculated "infinite" sortino score mean?

`
legend: 
--- = sale 

+++ = buy 

b = balance 

nw = net worth 

CS = calculating Sortino Rate 

reward = sortino rate, total_reward,  (length = min(self.current_step, self.forecast_len) 
`

DEBUG:__main__:Trainng length
INFO:__main__: Reward strategy : sortino
INFO:__main__: Input data file : data/small_corrected_hourly.csv
INFO:__main__: Parames db file : sqlite:///params.db
INFO:__main__: n_jobs          = 1
INFO:__main__: n_trials        = 30
INFO:__main__: n_test_episodes = 3
INFO:__main__: n_evaluations   = 4
INFO:__main__: Total number of records : 2612
INFO:__main__: Number of training recs : 2612
INFO:__main__: Num. of evaluation recs : 2089
INFO:__main__:forecast_length = 2
INFO:__main__:convidence_intv = 0.9407857951645567
INFO:__main__:n_steps         = 82
INFO:__main__:gamma           = 0.9910742592048303
INFO:__main__:learning_rate   = 0.0001463509003510531
INFO:__main__:ent_coef        = 8.4767762718177e-08
INFO:__main__:cliprange       = 0.9950675592874643
INFO:__main__:noptepochs      = 1
INFO:__main__:lam             = 0.9304853592231145

DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:--- $$$ 0.00
DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:--- $$$ 0.00
DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:--- $$$ 0.00
DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:                    b = 10000.00 nw = 10000.00
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:+++ $$$ 3333.33
DEBUG:env.BitcoinTradingEnv:                    b = 6666.67 nw = 6666.67
DEBUG:env.BitcoinTradingEnv:CS
DEBUG:env.BitcoinTradingEnv:                                             reward =  0 0   (2)
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:+++ $$$ 2222.22
DEBUG:env.BitcoinTradingEnv:                    b = 4444.44 nw = 4444.44
DEBUG:env.BitcoinTradingEnv:CS
DEBUG:env.BitcoinTradingEnv:                                             reward =  0 0   (2)
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:--- $$$ 5496.85
DEBUG:env.BitcoinTradingEnv:                    b = 9941.29 nw = 9941.29
DEBUG:env.BitcoinTradingEnv:CS
DEBUG:env.BitcoinTradingEnv:                                             reward =  0 0   (2)
DEBUG:env.BitcoinTradingEnv:

DEBUG:env.BitcoinTradingEnv:+++ $$$ 2485.32
DEBUG:env.BitcoinTradingEnv:                    b = 7455.97 nw = 7455.97
DEBUG:env.BitcoinTradingEnv:CS
DEBUG:env.BitcoinTradingEnv:                                             reward =  0 0   (2)
DEBUG:env.BitcoinTradingEnv:

`

**Comment by evanatyourservice** [comment]
Just add an "epsilon" to the denominator...
```
def sortino(self, returns):
    target = etc... # target returns

    numerator = np.mean(returns) - target

    returns = returns - target
    returns[returns > 0.0] = 0.0
    denominator = np.sqrt(np.mean(np.square(returns))) + 1.0e-8  # no divide by zero

    return numerator / denominator
```
Then it would be wise to normalize it somehow in the step function before it is sent to the algo, like taking the cube root or something. reward = self.sortino(returns)... reward = np.cbrt(reward)... ayyy
